src/popup/pages/home.tsx

```diff
@@ -138,7 +143,7 @@ const Home = () => {
                     )}
 
 
-                    <h3 className="text font-medium text-base leading-10">Tools:</h3>
+                    {renderToolsConditions.includes(true) && <h3 className="text font-medium text-base leading-10">Tools:</h3>}
 
                     <div className="tools flex flex-col gap-2">
```

Comment:
We can set a ref to this div and check the length of its children using `<REF>.current.children.length` before rendering the tools header. This way we won't have to maintain the `renderToolsConditions` array separately.

Reply:
I did initially try to do it that way, i was faced some difficulty and then i changed tactics.
although i feel we can benefit from such an array if there are other features we wish to make conditional similarly in the future.

Reply:
We can try it now. Try setting a ref, say `toolsRef`, for the div.
Replace the current condition with 
`{Boolean(toolsRef?.current?.children?.length) && ...Render the "Tools" header }`